packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -444,13 +444,22 @@ async function doEmit(
             emitterVersion: e.version ?? (await npmUtil.loadNpmPackage(e.package))?.version,
           });
         });
-        const compileResult = await compile(
-          cli,
-          mainTspFile,
-          emitters.map((e) => {
-            return { name: e.package, options: {} };
-          }),
-          false,
+        const compileResult = await telemetryClient.doOperationWithOperationDetailTelemetry(
```

Comment:
I think you just need to call logOperationDetailTelemetry to record the start and end time of compiler. this makes things unnecessary complex here.

Reply:
@RodgeFu same as above reply